google/resource_compute_target_ssl_proxy.go

```diff
@@ -0,0 +1,247 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/compute/v1"
+)
+
+func resourceComputeTargetSslProxy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceComputeTargetSslProxyCreate,
+		Read:   resourceComputeTargetSslProxyRead,
+		Delete: resourceComputeTargetSslProxyDelete,
+		Update: resourceComputeTargetSslProxyUpdate,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"backend_service": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"ssl_certificates": &schema.Schema{
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Schema{
+					Type:             schema.TypeString,
+					DiffSuppressFunc: compareSelfLinkOrResourceName,
+				},
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"proxy_header": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "NONE",
+			},
+
+			"project": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"proxy_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"self_link": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceComputeTargetSslProxyCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	sslCertificates, err := expandSslCertificates(d.Get("ssl_certificates").([]interface{}), d, config)
+	if err != nil {
+		return err
+	}
+
+	proxy := &compute.TargetSslProxy{
+		Name:            d.Get("name").(string),
+		Service:         d.Get("backend_service").(string),
+		ProxyHeader:     d.Get("proxy_header").(string),
+		Description:     d.Get("description").(string),
+		SslCertificates: sslCertificates,
+	}
+
+	log.Printf("[DEBUG] TargetSslProxy insert request: %#v", proxy)
+	op, err := config.clientCompute.TargetSslProxies.Insert(
+		project, proxy).Do()
+	if err != nil {
+		return fmt.Errorf("Error creating TargetSslProxy: %s", err)
+	}
+
+	err = computeOperationWait(config, op, project, "Creating Target Ssl Proxy")
+	if err != nil {
+		return err
+	}
+
+	d.SetId(proxy.Name)
+
+	return resourceComputeTargetSslProxyRead(d, meta)
+}
+
+func resourceComputeTargetSslProxyUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	d.Partial(true)
+
+	if d.HasChange("proxy_header") {
+		proxy_header := d.Get("proxy_header").(string)
```

Comment:
I hate to be that guy but this is using snake case rather than camel case.

It's totally readable to me but I imagine others might find it bad style